### PR TITLE
1984 entity header

### LINKED ISSUE #1984: [Sprint 5] Implement Project + Dataset headers design system
+ Release Cards + Headers + Forms feature branch

[Coteries Iteration 6 Figma Prototype](https://www.figma.com/proto/cmVH2YlFGz6uAv2EEH1sBZ/Renku---User-interface?page-id=1%3A2&node-id=29%3A756&viewport=1613%2C-2648%2C0.6&scaling=scale-down&starting-point-node-id=29%3A756)

Image

### PR DESCRIPTION
**Pull request by andre-code**
This PR add project header styles.

project and dataset header

### Testing cases
- Project header should show: title, description, visibility, update/creation time, author, tags, status icon, login button, description suggestion, slug, slug with link to forked project if applicable.
- Linked datasets section: a link to add a dataset if there are no datasets, a link to view the full list of datasets if there are more than 5 links.
- Linked section error retrieving datasets or projects: e.g. https://renku-ci-ui-2001.dev.renku.ch/projects/cramakri/dataset-test-test-project/datasets
- The header of the datasets should show: title, creation/publication time, author, keywords, option to delete, modify and import the dataset.
- Linked projects section: a message if there are no projects, a message indicating where to find the other projects when there are more than 5 links.

Closes #1984 

/deploy renku=1978-styles-cards-projects-and-datasets #persist

### CODE REVIEW

File: client/src/utils/components/entities/LinkedEntitiesByItemType.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  devAccess: boolean,
+  url: string
+}
+function LinkedEntitiesByItemType({ itemType, links, devAccess, url }: LinkedEntitiesByItemTypeProps) {
+  const dataByItem = {
+    project: {
+      title: "Linked datasets",
+      description: "Main datasets used in this project",
+      seeMore: "See more datasets...",
+      noLinks: "No dataset has been added yet.",
```
Comment (ciyer) [inline client/src/utils/components/entities/LinkedEntitiesByItemType.tsx:57]: I think ORDES should provide the text we show when there are no datasets, and also what interaction they want (show a button let users add a dataset?, do nothing?)

File: client/src/utils/components/entities/LinkedEntitiesByItemType.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      title: "Linked projects",
+      description: "Main projects that use this dataset",
+      seeMore: "... see more projects in the section below",
+      noLinks: "There are no projects using this dataset.",
+      icon: <Briefcase />
+    }
+  };
+  const stylesByItem = stylesByItemType(itemType);
+
+  const addDatasetLink = devAccess && itemType === "project" ?
```
Comment (ciyer) [inline client/src/utils/components/entities/LinkedEntitiesByItemType.tsx:70]: I think ORDES should specify if they want an interaction here and what text they want shown.

**Review by ciyer — CHANGES_REQUESTED** [review]
This looks very good, and already works well! 🎉 And the changes that are being done in parallel will make it work even better.

In testing, I found a few problems to fix:

- I cannot view datasets in Safari -- there is a crash in the TimeCaption (though it works in Chrome)
- On very old projects, ones that are not supported anymore, the "Linked Datasets" just shows the loading links spinner forever. (See https://renku-ci-ui-2001.dev.renku.ch/projects/cramakri/dataset-test-test-project/datasets)

image

- I do not understand what I'm supposed to be seeing in the second section of this dataset https://renku-ci-ui-2001.dev.renku.ch/projects/cramakri/dataset-test-project/datasets/image-dataset-3/

image

(That is a dataset with an image, but no files)

I also have a couple of comments in the code, but overall, this looks great and is close to being complete.

File: client/src/utils/components/TimeCaption.js [hunk 3]
```diff
@@ -34,21 +36,26 @@ function displayTimeFromDate(time) {
...
-    return <span className={className}>{caption} {displayTime}{endCaption}{endPunctuation}</span>;
-  }
+  const displayTime = displayTimeFromDate(time);
+  const endCaptionStyled = (endCaption) ? " " + endCaption : "";
+  if (!noCaption)
+    className = "time-caption " + className;
+
+  const ref = useRef(null);
+  const tooltip = showTooltip ?
+    <ThrottledTooltip target={ref} tooltip={Time.toIsoString(time)} /> : null;
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/utils/components/TimeCaption.js:50]: Yes, it has been fixed in the last commit

**Comment by andre-code** [comment]
Hi @ciyer thanks for your review and suggestions. I have fixed all the crash cases in safari and also included the option to show an error message when the linked dataset list or project cannot be retrieved.
Everything about the product (texts and expected behavior) I will confirm to ORDES.

**Review by ciyer — APPROVED** [review]
Looks really great! I am approving. I know there are some questions with ORDES that might still be open. If you want, you can wait for those to be resolved, or you can merge and then make the changes in a later PR.

**Pull request merged**